api/v1beta1/imageupdateautomation_types.go

```diff
@@ -29,7 +30,7 @@ type ImageUpdateAutomationSpec struct {
 	// SourceRef refers to the resource giving access details
 	// to a git repository.
 	// +required
-	SourceRef SourceReference `json:"sourceRef"`
+	SourceRef dependency.CrossNamespaceDependencyReference `json:"sourceRef"`
```

Comment:
You need to create a type like this one https://github.com/fluxcd/kustomize-controller/blob/main/api/v1beta2/reference_types.go with `+kubebuilder:validation:Enum=GitRepository`

Reply:
Using `CrossNamespaceDependencyReference` breaks the API, but because this controller lacks any e2e tests such changes could make it into release. We should get the IRC and IAC in line with all the others controllers and write proper e2e tests.